### PR TITLE
[mqtt.homeassistant] Implement WaterHeater

### PR DESCRIPTION
**Pull request by ccutrer**

### CODE REVIEW

**Comment by ccutrer** [comment]
This is built on top of #17853, and once that is merged, this one will need to be rebased (including any additional changes that PR has, to ensure a clean rebase).

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/resources/OH-INF/thing/homeassistant-channels.xml [hunk 1]
```diff
@@ -66,6 +66,12 @@
 		<config-description-ref uri="channel-type:mqtt:ha-channel"/>
 	</channel-type>
 
+	<channel-type id="ha-temperature">
+		<item-type unitHint="%">Number:Temperature</item-type>
```
Comment (lsiepel) [inline bundles/org.openhab.binding.mqtt.homeassistant/src/main/resources/OH-INF/thing/homeassistant-channels.xml:70]: The unitHint is wrong here.

**Review by lsiepel — CHANGES_REQUESTED** [review]
1 comment, otherwise LGTM

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/resources/OH-INF/thing/homeassistant-channels.xml (hunk 1 above)
Comment (ccutrer) [reply bundles/org.openhab.binding.mqtt.homeassistant/src/main/resources/OH-INF/thing/homeassistant-channels.xml]: 🤦 . I still had Humidity on my mind.

**Pull request merged**